core/src/main/java/org/apache/iceberg/deletes/Deletes.java

```diff
@@ -73,6 +74,16 @@ public static <T> CloseableIterable<T> filter(CloseableIterable<T> rows, Functio
     return filter.filter(rows);
   }
 
+  public static <T> CloseableIterable<T> marker(CloseableIterable<T> rows, Function<T, Long> rowToPosition,
```

Comment:
nit: `marker` is a noun,  I think here should use a verb. Maybe `filter` still fits since it is still filtering.